### PR TITLE
[docs] Follower reads in YSQL

### LINKED ISSUE #10506: [Docs] Add docs for YSQL follower reads
### Description

internal functional spec: https://docs.google.com/document/d/1Pb-ZxegHKwFxvMBfcRoOIB2ldZluKAYkCiqM0uT4o2A/edit#heading=h.mior1do4w7bm

cc @amitanandaiyer for any content asks

### PR DESCRIPTION
**Pull request by aishwarya24**
Resolves #10506

Updated Build preview : https://deploy-preview-10670--infallible-bardeen-164bc9.netlify.app/latest/explore/multi-region-deployments/follower-reads-ysql/ 
- Added the document to both latest and stable. 
- The document currently includes an introduction, details about the session variables and various examples that demonstrates follower reads.
- The default for follower reads now points to YSQL.
 
@netlify /latest/explore/multi-region-deployments/follower-reads-ysql/

### CODE REVIEW

**Comment by bmatican** [comment]
Added Amit, if you wanted a technical review of the content as well.

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md [hunk 1]
```diff
@@ -9,25 +9,245 @@ menu:
...
       <i class="icon-cassandra" aria-hidden="true"></i>YCQL</a>
-  </li>  
-
+  </li>
 </ul>
 
-YSQL follower reads support is currently in Beta. 
+## Introduction
+
+YugabyteDB requires reading from the **leader** to read the latest data. However, for applications that do not require the latest data, and/or are working with unchanging data; the cost of contacting a potentially remote leader to fetch the data may be wasteful. The application could benefit with better latency by reading from a **replica** that may be closer to the client.
```
Comment (amitanandaiyer) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:31]: `benefit with` sounds off. Perhaps a native speaker may check if this is the right usage.

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md [hunk 2]
```diff
@@ -9,25 +9,245 @@ menu:
...
+```
+
+This next two examples use follower reads since **pg_hint** can be used during PREPARE and CREATE FUNCTION to do follower reads.
+
+```sql
+set session characteristics as transaction read write;
+set yb_read_from_followers = true;
+
+PREPARE select_stmt(text) AS
+/*+ Set(transaction_read_only on)*/
```
Comment (amitanandaiyer) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:114]: Do we need a space before */

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md [hunk 3]
```diff
@@ -9,25 +9,245 @@ menu:
...
+
+- *yb_follower_read_staleness_ms* : Defaults to 30 seconds.
+
+### Expected behavior
+
+The table describes what the expected behavior is when a read happens from a follower.
+
+| Conditions | Expected behavior |
+| --- | --- |
+| yb_read_from_followers is true AND <br> (Transaction is marked read-only <br>OR<br> Single-statement-select can be inferred to be read-only) <br> | Read will happen from the closest follower|
```
Comment (amitanandaiyer) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:55]: single statement select inferrence isn't landed yet.

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md [hunk 4]
```diff
@@ -9,25 +9,245 @@ menu:
...
+|yb_read_from_followers is false OR <br> Transaction/statement is not read only | Read will happen from the leader |
+
+### Read from follower conditions
+
+- If the follower’s safe-time is larger than the read-time thus chosen, the follower may serve the read without any delay.
```
Comment (kmuthukk) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:60]: "thus chosen" -- chosen how? This hasn't been explained before on this page.

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md [hunk 5]
```diff
@@ -9,25 +9,245 @@ menu:
...
+
+- If the follower is not yet caught up to the specified timestamp, the read will be redirected to a different replica transparently from the end-user. The end user may see a slight increase in latency depending on the location of the replica which satisfies the read.
```
Comment (kmuthukk) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:62]: "the specified timestamp" -- which timestamp? Again, this concept has not been explained.

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md [hunk 6]
```diff
@@ -9,25 +9,245 @@ menu:
...
+  - The current behavior is that a follower/replica would wait for the requested timestamp to become safe. This behavior shall be changed to respond back to the `ybclient/pggate` immediately, so that the request may be routed to a different server that has the data.
```
Comment (kmuthukk) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:63]: This discussion is not relevant to external documentation, @amitanandaiyer .

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md [hunk 7]
```diff
@@ -9,25 +9,245 @@ menu:
...
+  - The current redirection policy is to directly go to the leader if the follower read is rejected by a follower. This behavior could be also be changed going forward, to explore other replicas before going to the leader.
+  - If the request reaches the leader, it will no longer reject/redirect the request.
+
+### Read-only transaction conditions
+
+Regarding marking the transaction as read only, a user can do one of the following:
+
+- `SET TRANSACTION READ ONLY` : Applies only to the current transaction block.
```
Comment (kmuthukk) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:71]: Minor: No space before the : 

Ditto on the next two lines also.

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md [hunk 8]
```diff
@@ -9,25 +9,245 @@ menu:
...
+
+- If the follower is not yet caught up to the specified timestamp, the read will be redirected to a different replica transparently from the end-user. The end user may see a slight increase in latency depending on the location of the replica which satisfies the read.
+  - The current behavior is that a follower/replica would wait for the requested timestamp to become safe. This behavior shall be changed to respond back to the `ybclient/pggate` immediately, so that the request may be routed to a different server that has the data.
+  - The current redirection policy is to directly go to the leader if the follower read is rejected by a follower. This behavior could be also be changed going forward, to explore other replicas before going to the leader.
+  - If the request reaches the leader, it will no longer reject/redirect the request.
+
+### Read-only transaction conditions
+
+Regarding marking the transaction as read only, a user can do one of the following:
+
```
Comment (kmuthukk) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:70]: How come there's no mention of pg_hint syntax in this section as another way of achieving same thing?

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md [hunk 9]
```diff
@@ -9,25 +9,245 @@ menu:
...
+```
+
+```output
+ k  | v
+----+----
+ k1 | v1
+(1 row)
+```
+
+This next two examples use follower reads since **pg_hint** can be used during PREPARE and CREATE FUNCTION to do follower reads.
```
Comment (kmuthukk) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:107]: Can't pg_hint method be used for a top-level/standalone SELECT statement? Does it have to be only via PREPARE or CREATE FUNCTIONA?

**Review by kmuthukk — COMMENTED** [review]
Left some comments/questions on this.

Thx. CC: @amitanandaiyer

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md [hunk 10]
```diff
@@ -9,25 +9,245 @@ menu:
...
+```
+
+```output
+ k  |  v
+----+------
+ k1 | v1
+(1 row)
+```
+
+The final example demonstrates that SELECTS error out in read-only transactions.
```
Comment (kmuthukk) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:242]: The wording needs to be improved here. 

It is not that SELECTs error out in a read-only transaction. It has more do with the fact that SELECT INTO is not a read-only transaction.

And this example has nothing to do with follower reads. Because we would get the same error even if follower reads was off.

So what's the point  of this example?

CC: @amitanandaiyer , @aishwarya24 ?

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md [hunk 11]
```diff
@@ -9,25 +9,245 @@ menu:
...
+ k  |  v
+----+------
+ k1 | v1
+(1 row)
+```
+
+```sql
+set yb_follower_read_staleness_ms = 5000;
+
+select * from t where k = 'k1';   /* 5s old value */
```
Comment (kmuthukk) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:207]: Is this 5s old value or up to 5s old value?

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md [hunk 12]
```diff
@@ -9,25 +9,245 @@ menu:
...
+
+- *yb_read_from_followers* : This session variable controls whether read from followers is enabled.
+
+- *yb_follower_read_staleness_ms* : This session variable defines the maximum staleness that the user is willing to tolerate.
+
+### Default behavior
+
+- *yb_read_from_followers* : Defaults to false.
+
+- *yb_follower_read_staleness_ms* : Defaults to 30 seconds.
```
Comment (polarweasel) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:47]:
```suggestion
Two session variables control the behavior of follower reads:

- `yb_read_from_followers` controls whether reading from followers is enabled. Default is false.

- `yb_follower_read_staleness_ms` sets the maximum allowable staleness. Default is 30 seconds.
```

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md (hunk 3 above)
Comment (polarweasel) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:55]:
```suggestion
| Conditions | Expected behavior |
| :--------- | :---------------- |
| yb_read_from_followers is true AND transaction is marked read-only | Read happens from the closest follower |
```

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md [hunk 13]
```diff
@@ -9,25 +9,245 @@ menu:
...
+
+### Expected behavior
+
+The table describes what the expected behavior is when a read happens from a follower.
+
+| Conditions | Expected behavior |
+| --- | --- |
+| yb_read_from_followers is true AND <br> (Transaction is marked read-only <br>OR<br> Single-statement-select can be inferred to be read-only) <br> | Read will happen from the closest follower|
+|yb_read_from_followers is false OR <br> Transaction/statement is not read only | Read will happen from the leader |
```
Comment (polarweasel) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:56]:
```suggestion
| yb_read_from_followers is false OR transaction/statement is not read-only | Read happens from the leader |
```

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md (hunk 3 above)
Comment (polarweasel) [reply docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:55]: My suggestion below addresses this, and we should add a separate row to the table when it does land. There's no need for complex logic in a reference table. :)

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md (hunk 4 above)
Comment (polarweasel) [reply docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:60]: Do we actually mean "If the follower's data is fresher than the client's yb_follower_read_staleness_ms value" here?

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md (hunk 7 above)
Comment (polarweasel) [reply docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:71]: Even better: no colon, lowercase the word that follows: "`SET TRANSACTION READ ONLY` applies only to the..."

Comment (polarweasel) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:71]:
```suggestion
- `SET TRANSACTION READ ONLY` applies only to the current transaction block.
```

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md [hunk 14]
```diff
@@ -9,25 +9,245 @@ menu:
...
+  - The current redirection policy is to directly go to the leader if the follower read is rejected by a follower. This behavior could be also be changed going forward, to explore other replicas before going to the leader.
+  - If the request reaches the leader, it will no longer reject/redirect the request.
+
+### Read-only transaction conditions
+
+Regarding marking the transaction as read only, a user can do one of the following:
+
+- `SET TRANSACTION READ ONLY` : Applies only to the current transaction block.
+- `SET SESSION CHARACTERISTICS AS TRANSACTION READ ONLY` : Applies the read-only setting for all statements and transaction blocks that follow.
+- `SET default_transaction_read_only = TRUE` : Applies the read-only setting for all statements and transaction blocks that follow.
```
Comment (polarweasel) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:73]:
```suggestion
- `SET SESSION CHARACTERISTICS AS TRANSACTION READ ONLY` applies the read-only setting to all statements and transaction blocks that follow.
- `SET default_transaction_read_only = TRUE` applies the read-only setting to all statements and transaction blocks that follow.
```

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md (hunk 11 above)
Comment (aishwarya24) [reply docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:207]: Will change this to "up to 5s old value" wherever applicable.

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md [hunk 15]
```diff
@@ -9,25 +9,245 @@ menu:
...
+```output
+ k  |  v
+----+------
+ k1 | v1+1
+(1 row)
+```
+
+```sql
+postgres=# set yb_follower_read_staleness_ms = 25000;
+postgres=# select * from t where k = 'k1';   /* 25s old value */
```
Comment (polarweasel) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:232]: Why are these Postgres prompts and not ysqlsh?

**Review by polarweasel — CHANGES_REQUESTED** [review]

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md (hunk 11 above)
Comment (polarweasel) [reply docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:207]: (Same for the 15 and 25 versions :) )

**Comment by polarweasel** [comment]
The FR page(s) clearly need more contextual information, but in the interests of getting _something_ documented since 2.11 is already out, I would suggest we address the errors and other things we can do reasonably quickly, and then follow up with more thorough explanations soon, like next week. @kmuthukk and @amitanandaiyer does that work for you?

**Comment by kmuthukk** [comment]
Sure @polarweasel .

@aishwarya24 -- when the round of comments have been incorporated, please let me know the netlify preview url, and will take a look again,

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md (hunk 4 above)
Comment (amitanandaiyer) [reply docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:60]: I'd say rewite this as:

If the follower's safe-time is at least <current_time> - <staleness>, 

If the follower is not yet caught up to <current_time> - <staleness> ...

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md (hunk 5 above)
Comment (amitanandaiyer) [reply docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:62]: If the follower is not yet caught up to `<current_time> - <staleness> `...

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md (hunk 6 above)
Comment (amitanandaiyer) [reply docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:63]: Yes. The 3 sub bullets:
Screen Shot 2021-11-30 at 10 32 08 AM

may be removed from the external documentation.

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md (hunk 8 above)
Comment (amitanandaiyer) [reply docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:70]: updated the FS.

Can you pls also add

* SET transaction_read_only = TRUE (applies to the current transaction block)
* /*+ Set(transaction_read_only true) */ SELECT (applies only to the current transaction/select statement)
       
here.

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md (hunk 9 above)
Comment (amitanandaiyer) [reply docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:107]: Yes. the pg_hint method can also be used for stand_alone statements.

The important point to highlight there is that for Prepare/Functions, the pg_hint needs to be applied at the prepare/function-definition stage. Not at the execute stage.

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md (hunk 10 above)
Comment (amitanandaiyer) [reply docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:242]: Yes. That's true. We could just remove this example.

It was added to the FS to clarify the discussion regarding how we'd prevent/safegurard against a select statement  having non-read-only side-effects. 

Doesn't need to be part of the user documentation.

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md (hunk 9 above)
Comment (aishwarya24) [reply docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:107]: I will include it as a note.

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md [hunk 16]
```diff
@@ -9,25 +9,247 @@ menu:
...
+```
+
+```output
+ k  | v
+----+----
+ k1 | v1
+(1 row)
+```
+
+The following examples use follower reads since **pg_hint** can be used during PREPARE, CREATE FUNCTION, and SELECT to do follower reads.
```
Comment (kmuthukk) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:100]: replace all occurrences of pg_hint with pg_hint_plan, as I believe that's the correct word.

since **pg_hint**
-->
since the **pg_hint_plan** mechanism

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md [hunk 17]
```diff
@@ -9,25 +9,247 @@ menu:
...
+Replicas may not be completely up to date with all updates, so this design may respond with stale data. You can specify how much staleness the application can tolerate. When enabled, read-only operations may be handled by the closest replica, instead of having to go to the leader.
+
+## Surface area and usage
+
+### Surface area
+
+Two session variables control the behavior of follower reads:
+
+- `yb_read_from_followers` controls whether reading from followers is enabled. Default is false.
+- `yb_follower_read_staleness_ms` sets the maximum allowable staleness. Default is 30 seconds.
```
Comment (kmuthukk) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:42]: is 30 seconds.
-->
is 30000 (i.e. 30 secs).

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md [hunk 18]
```diff
@@ -9,25 +9,247 @@ menu:
...
+
+### Read from follower conditions
+
+- If the follower’s safe-time is at least `<current_time> - <staleness>`, then the follower may serve the read without any delay.
+
+- If the follower is not yet caught up to `<current_time> - <staleness>`, then the read will be redirected to a different replica transparently from the end-user. The end user may see a slight increase in latency depending on the location of the replica which satisfies the read.
+
+### Read-only transaction conditions
+
+Regarding marking the transaction as read only, a user can do one of the following:
```
Comment (kmuthukk) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:61]: Regarding marking the transaction
-->
To mark a transaction

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md [hunk 19]
```diff
@@ -9,25 +9,247 @@ menu:
...
+
+- `SET TRANSACTION READ ONLY` applies only to the current transaction block.
+- `SET SESSION CHARACTERISTICS AS TRANSACTION READ ONLY` applies the read-only setting to all statements and transaction blocks that follow.
+- `SET default_transaction_read_only = TRUE` applies the read-only setting to all statements and transaction blocks that follow.
+- `/*+ Set(transaction_read_only true) */ SELECT` applies only to the current transaction/select statement.
```
Comment (kmuthukk) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:66]: Use [pg_hint_plan](https://docs.yugabyte.com/latest/explore/query-1-performance/pg-hint-plan/) mechanism	to embed the hint along with the SELECT statement. For example,` /*+ Set(transaction_read_only true) */ SELECT ...`  applies only	to the current SELECT  statement.

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md [hunk 20]
```diff
@@ -9,25 +9,247 @@ menu:
...
+
+```output
+ k  | v
+----+----
+ k1 | v1
+(1 row)
+```
+
+```sql
+set session characteristics as transaction read write;
```
Comment (kmuthukk) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:143]: Why is this line needed?

And overall the simple SELECT with the embedded hint should be the first example for pg_hint_plan, instead of the more uncommon cases of PREPARE and CREATE FUNCTION. The flow needs to be simple to advanced rather than reverse.

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md [hunk 21]
```diff
@@ -9,25 +9,247 @@ menu:
...
+ k  | v
+----+----
+ k1 | v1
+(1 row)
+```
+
+The following examples use follower reads since **pg_hint** can be used during PREPARE, CREATE FUNCTION, and SELECT to do follower reads.
+
+{{< note title="Note" >}}
+The pg_hint needs to be applied at the prepare/function-definition stage and not at the `execute` stage.
```
Comment (kmuthukk) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:103]: The pg_hint needs to 
-->
The pg_hint_plan hint needs to

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md [hunk 22]
```diff
@@ -9,25 +9,247 @@ menu:
...
+ k | v | k | v
+---+---+---+---
+ 3 | 6 | 2 | 6
+(1 row)
+```
+
+The following examples demonstrate **staleness** after enabling the `yb_follower_read_staleness_ms` property.
+
+```sql
+set session characteristics as transaction read write;
```
Comment (kmuthukk) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:177]: @amitanandaiyer , @aishwarya24 :

between 'v1' becoming 'v1+1' don't you need a 10sec sleep to illustrate the point. I don't see that being explicitly mentioned. Would be good to show the INSERT of 'v1' and a sleep 10s after that also, in my opinion.

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md [hunk 23]
```diff
@@ -9,25 +9,247 @@ menu:
...
+UPDATE t SET  v = 'v1+1' where k = 'k1';
+/* sleep 10s */
+
+UPDATE t SET  v = 'v1+2' where k = 'k1';
+/* sleep 10s */
+
+select * from t where k = 'k1';
+```
+
+```output
```
Comment (kmuthukk) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:198]: Might be worth adding some commentary along with..

For example, here I would say:

This selects the latest version of the row because the transaction setting for the session is `read write`.

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md (hunk 20 above)
Comment (amitanandaiyer) [reply docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:143]: It is just to highlight that pg_hint can be used in a read_write session to override the read only property.

We don't need this line per se. The example would work even without this line. So we can remove this, and otehr places in line 125 and 107.

Also, agree with moving this example above line 107. We can show this example, then the PREPARE example and then the CREATE FUNCTION example.

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md (hunk 22 above)
Comment (amitanandaiyer) [reply docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:177]: So the assumption here was that the initial state of the row was `v1` (as depicted by the select). With the write being made (at least 5+ seconds) in the past.

we could add it here, to make it more explicitly, if the default assumption is that the table/row is empty.

So replace the

```
insert into t values ('k1', 'v1')
/* sleep 10s */
```

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md (hunk 20 above)
Comment (aishwarya24) [reply docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:143]: Thank you ! I'll make the change accordingly.

File: docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md [hunk 24]
```diff
@@ -39,7 +39,7 @@ Replicas may not be completely up to date with all updates, so this design may r
 Two session variables control the behavior of follower reads:
 
 - `yb_read_from_followers` controls whether reading from followers is enabled. Default is false.
-- `yb_follower_read_staleness_ms` sets the maximum allowable staleness. Default is 30 seconds.
+- `yb_follower_read_staleness_ms` sets the maximum allowable staleness. Default is 30000(30 seconds).
```
Comment (kmuthukk) [inline docs/content/latest/explore/multi-region-deployments/follower-reads-ysql.md:42]: there should be a space before (

File: docs/content/stable/explore/multi-region-deployments/follower-reads-ysql.md [hunk 25]
```diff
@@ -39,7 +39,7 @@ Replicas may not be completely up to date with all updates, so this design may r
 Two session variables control the behavior of follower reads:
 
 - `yb_read_from_followers` controls whether reading from followers is enabled. Default is false.
-- `yb_follower_read_staleness_ms` sets the maximum allowable staleness. Default is 30 seconds.
+- `yb_follower_read_staleness_ms` sets the maximum allowable staleness. Default is 30000(30 seconds).
```
Comment (kmuthukk) [inline docs/content/stable/explore/multi-region-deployments/follower-reads-ysql.md:42]: space before (

**Review by kmuthukk — COMMENTED** [review]
Why do we see both latest/explore and stable/explore version of changes here?

My understanding is that follower reads is right now only available in 2.11 and not yet in a stable release.

CC: @ymahajan

**Comment by aishwarya24** [comment]
@kmuthukk , Confirmed with @polarweasel and have applied the changes only to latest/explore. Thank you.

**Review by kmuthukk — APPROVED** [review]
Thx!

**Comment by polarweasel** [comment]
Sent you a few slack comments. Probably GTG for now, but will need some followup adjustments.
